website/docs/contributing/ADRs/front-end/sdk-generator.md

```diff
@@ -0,0 +1,33 @@
+---
+title: "ADR: OpenAPI SDK Generator"
+---
+
+## Background
+
+Our current frontend setup we have a lot of code that can be generated out of OpenAPI schema. Types have not been updated in a while, and even some new features rely on hard-coded types instead of auto-generated files in `src/openapi`. Fetchers and actions in the frontend involves custom-built code that is grouped in the `src/hooks/api` folder. There is a separation between getters and actions. Getters use the SWR library. API actions (POST/PUT/DELETE) are grouped by feature and are exposed to components reliant on `useAPI` hook.
+
+SDK generated out of it will be better then what we have right now. It will help to reduce the risk of duplication and inconsistencies in our SDK, as it will be generated from a single source of truth.
+
+## Decisions
+
+- We will use the [Orval](https://orval.dev/) package to generate the typescript types for our SDK.
+- We will consider using Orval to generate the HTTP getters and actions for our SDK in the future, but will first carefully test this approach in new features under development to weed out edge cases.
+- We will deprecate `src/interfaces` related to API calls and use `src/openapi` models instead.
+
+### Advantages
+
+- We retain the flexibility of previous solution, because we can implement our own _fetcher_ function, and substitute _response_ and _error_ type generics. See https://orval.dev/guides/custom-client
+- Orval supports `anyOf` and `oneOf` schema, which the previous generator did not support.
+- If we decide to use Orval to generate the HTTP getters and actions for our SDK, it will reduce the amount of boilerplate code required when working with new APIs.
+- We can always revert to writing some API calls by hand if this approach is not flexible enough.
+
+### Concerns
```

Comment:
In terms of concerns this twitter conversation summarizes it nicely: https://twitter.com/gicagalbenu/status/1494933892310212611